src/pages/Models/Details/ModelDetails.js

```diff
@@ -266,54 +201,37 @@ const ModelDetails = () => {
     }
   }, [dispatch, modelUUID, modelStatusData]);
 
-  const handleAppRowClick = (e, app) => {
-    const currentApp = cloneDeep(app);
-    currentApp.name = e.currentTarget.dataset.app;
-    setSlidePanelData({ currentApp });
-    setFilterByApp(currentApp.name);
+  const handleAppRowClick = (e) => {
+    setQuery({ panel: "apps", entity: e.currentTarget.dataset.app });
+  };
+
+  const closePanel = () => {
+    setQuery({ panel: undefined, entity: undefined });
   };
 
   const applicationTableRows = useMemo(
     () =>
       generateApplicationRows(modelStatusData, handleAppRowClick, baseAppURL),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can you add a comment why this is required? Usually when this is required something else is wrong :)

Reply:
Refactored useMemo so this exception is no longer required.